### ISSUE TITLE
log_driver: journald not supported

### ISSUE THREAD

**Issue by jeremyd** [docker/compose#2021]
I want to use docker-compose to launch docker 1.7.1 with the log_driver for journald. However when I specify log_driver: journald in the yaml config it says "ValueError: LogConfig.type must be one of (json-file, syslog, none)"

**Comment by jeremyd** [comment]
full traceback

```
Traceback (most recent call last):
  File "/usr/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.4.0', 'console_scripts', 'docker-compose')()
  File "/usr/lib/python2.7/site-packages/compose/cli/main.py", line 39, in main
    command.sys_dispatch()
  File "/usr/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 21, in sys_dispatch
    self.dispatch(sys.argv[1:], None)
  File "/usr/lib/python2.7/site-packages/compose/cli/command.py", line 27, in dispatch
    super(Command, self).dispatch(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 24, in dispatch
    self.perform_command(*self.parse(argv, global_options))
  File "/usr/lib/python2.7/site-packages/compose/cli/command.py", line 59, in perform_command
    handler(project, command_options)
  File "/usr/lib/python2.7/site-packages/compose/cli/main.py", line 495, in up
    timeout=timeout
  File "/usr/lib/python2.7/site-packages/compose/project.py", line 265, in up
    timeout=timeout
  File "/usr/lib/python2.7/site-packages/compose/service.py", line 369, in execute_convergence_plan
    do_build=do_build,
  File "/usr/lib/python2.7/site-packages/compose/service.py", line 270, in create_container
    previous_container=previous_container,
  File "/usr/lib/python2.7/site-packages/compose/service.py", line 643, in _get_container_create_options
    one_off=one_off)
  File "/usr/lib/python2.7/site-packages/compose/service.py", line 656, in _get_container_host_config
    config=options.get('log_opt', None)
  File "/usr/lib/python2.7/site-packages/docker/utils/types.py", line 27, in __init__
    ', '.join(self.types._values)
ValueError: LogConfig.type must be one of (json-file, syslog, none)
```

**Comment by jeremyd** [comment]
Looks like this is already in upstream https://github.com/docker/docker-py/blob/master/docker/utils/types.py

**Comment by mnowster** [comment]
I believe this will be fixed when we release 1.5 of docker-compose as it relies on a few things in docker-py which will also be due a release, then we can update the dependencies safely.

Removing the constraints on log_driver's is also covered by this issue: https://github.com/docker/compose/issues/1869
